benchmarks/src/test/scala/org/ergoplatform/nodeView/state/TransactionExecutionBenchmark.scala

```diff
@@ -0,0 +1,51 @@
+package org.ergoplatform.nodeView.state
+
+import org.ergoplatform.Utils
+import org.ergoplatform.Utils.BenchReport
+import org.ergoplatform.modifiers.mempool.ErgoTransaction
+import org.ergoplatform.nodeView.NVBenchmark
+import org.ergoplatform.utils.{HistoryTestHelpers, RandomWrapper}
+import scorex.core.validation.ValidationResult.Valid
+import scorex.db.ByteArrayWrapper
+
+import scala.collection.mutable
+import scala.util.Try
+
+object TransactionExecutionBenchmark extends HistoryTestHelpers with NVBenchmark {
+
+  val WarmupRuns = 3
+
+  def main(args: Array[String]): Unit = {
+
+    val startTs = System.currentTimeMillis()
+
+    val bh = BoxHolder(genesisBoxes)
+    val txs = (1 to 5000).foldLeft(mutable.WrappedArray.empty[ErgoTransaction]) { case (txAcc, _) =>
+      val (transactions, _) = validTransactionsFromBoxes(10000, bh.boxes.values.toVector, new RandomWrapper)
+      val allBoxIds = bh.boxes.keys.toSet
+      val txsFromBoxesOnly = transactions.filter { tx =>
+        tx.inputs.map(i => ByteArrayWrapper(i.boxId)).forall(allBoxIds.contains) &&
+          tx.dataInputs.map(i => ByteArrayWrapper(i.boxId)).forall(allBoxIds.contains)
+      }
+      txAcc ++ txsFromBoxesOnly
```

Comment:
For mutable fold state you can use ArrayBuilder, which has amortized O(1) append operation.
Whereas here you have O(n^2) due repeated copying.